tools/dmi/extract_dmi_states.ps1

```
@@ -0,0 +1,36 @@
+            $compressedLen = $dataLen - ($compressedStart - $dataStart)
+            $compressed = [byte[]]::new($compressedLen)
+            [Array]::Copy($bytes, $compressedStart, $compressed, 0, $compressedLen)
+            try {
+                $ms = [System.IO.MemoryStream]::new($compressed)
+                $ds = [System.IO.Compression.DeflateStream]::new($ms, [System.IO.Compression.CompressionMode]::Decompress)
+                $out = [System.IO.MemoryStream]::new()
+                $ds.CopyTo($out)
+                $ds.Close()
+                $text = [System.Text.Encoding]::UTF8.GetString($out.ToArray())
+                [regex]::Matches($text, 'state\s*=\s*([^\r\n]+)') | ForEach-Object { $_.Groups[1].Value.Trim() }
+            } catch {}
```

PSScriptAnalyzer (1.25.0)

[warning] Empty catch block is used. Please use Write-Error or throw statements in catch blocks.

(PSAvoidUsingEmptyCatchBlock)